snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/cli/SnomedCommand.java

```diff
@@ -142,6 +146,79 @@ public void run(CommandLineStream out) {
 		}
 	}
 	
+	@picocli.CommandLine.Command(
+			name = "identifiers",
+			header = "Collects SNOMED CT core component identifiers",
+			description = "Collect SNOMED CT identifiers"
+			)
+	private static final class IdentifiersCommand extends Command {
+				
+		private static final Set<String> ALL_SCTID_STATUSES = FluentIterable.from(IdentifierStatus.values()).transform(IdentifierStatus::getSerializedName).toSet();
+				
+		@Option(names = { "-s", "--status" }, description = "The SctId status to filter for. Can be Available, Reserved, Assigned, or Published.", defaultValue = "All", required = false)
+		String status="All";
+		
+		@Parameters(paramLabel = "PATH", description = "The absolute path of the output folder")
+		String path;
+		
+		
+		@Override
+		public void run(CommandLineStream out) {
+			ISnomedIdentifierService identifierService = getContext().service(ISnomedIdentifierService.class);
+			
+			if (!(identifierService instanceof DefaultSnomedIdentifierService)) {
+				out.println("Can't run command without DefaultSnomedIdentifierService");
+				return;
+			}
+			
+			DefaultSnomedIdentifierService defaultIdentifierService = (DefaultSnomedIdentifierService) identifierService;
+			
+			Set<String> statuses = Sets.newHashSet();
+			
+			if (ALL_SCTID_STATUSES.contains(status)) {
+				statuses.add(status);
+			} else {
+				statuses.addAll(ALL_SCTID_STATUSES);
+			}
+			
+			defaultIdentifierService.store.read( searcher -> {
+				statuses.forEach( status -> {
+					Query<String> idQuery = Query.select(String.class)
+							.from(SctId.class)
+							.fields("sctid")
+							.where(SctId.Expressions.status(status))
+							.limit(100_000)
+							.build();
+					
+					File idReport = new File(String.format("%s/%sIds.txt", path, status));
+					try (FileOutputStream outputStream = new FileOutputStream(idReport)) {
+						try (OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream, "UTF-8")) {
+							searcher.scroll(idQuery).forEach( hits -> {
+								hits.forEach( id -> {
+									try {
+										outputStreamWriter.append(String.format("%s\n", id));
+									} catch (Exception e) {
+										handleException(e, out);
```

Comment:
While this handles the exception and logs it, it continues the execution of the ID extraction. Is that intended?

Reply:
I'm still not seeing any change that would allow this comment to be resolved.